### PR TITLE
EC2 task networking for Windows: Build vpc-eni CNI plugin for Windows

### PR DESCRIPTION
**Pull request by rawahars**
### Summary

The changes introduced in this PR are-

- Build the `vpc-eni` plugin for Windows
- Updated the retry values during setup NS 

### Implementation details

The changes were implemented using -

- Build `vpc-eni` plugins during the agent build
- Changed the max retries to 5 and retryMultiplier to 2. These changes are based on the practical observations.

### Testing

A custom binary of the agent was tested.
New tests cover the changes: 
No new changes have been introduced.
### Description for the changelog

Build vpc-eni plugin for ECS Windows task networking
### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -20,4 +20,4 @@ jobs:
         $Env:GOPATH = "$Env:GITHUB_WORKSPACE"
         cd "$Env:GITHUB_WORKSPACE"
         cd "src/github.com/aws/amazon-ecs-agent"
-        go test -v -race -tags unit -timeout 40s ./agent/...
+        go test -race -tags unit -timeout 40s ./agent/...
```
Comment (vsiddharth) [inline .github/workflows/windows.yml:23]: Why did we remove the `-v` ?

File: agent/ecscni/types_windows.go [hunk 2]
```diff
@@ -28,12 +28,16 @@ const (
 	ECSVPCENIPluginExecutable = "vpc-eni.exe"
 	// ECSBridgeNetworkName is the name of the HNS network used as ecs-bridge.
 	ECSBridgeNetworkName = "nat"
-	// Constants for creating backoff while retrying setupNS.
+)
+
+var (
+	// Values for creating backoff while retrying setupNS.
+	// These have not been made constant so that we can inject different values for UTs.
```
Comment (vsiddharth) [inline agent/ecscni/types_windows.go:35]: nit: please expand `UTs`

**Review by vsiddharth — COMMENTED** [review]
Changes look good. Can you please look into the test failures?

File: .github/workflows/windows.yml (hunk 1 above)
Comment (rawahars) [reply .github/workflows/windows.yml:23]: By removing this flag, we are able to remove the excess noise from the UTs during PR. We can very easily pin point the UT which failed and the corresponding error. For example - https://github.com/aws/amazon-ecs-agent/runs/2786834601?check_suite_focus=true

Also, this ensures that the behavior is same for Linux and Windows UT during PR build.

Comment (vsiddharth) [reply .github/workflows/windows.yml:23]: Nice 👍

**Review by vsiddharth — APPROVED** [review]
Neat 🚀

**Review by fenxiong — APPROVED** [review]

**Review by angelcar — APPROVED** [review]

**Pull request merged**